tools/api.py

```diff
@@ -1059,7 +1059,7 @@ def add_cpp_class(self) -> None:
         throw std::runtime_error("${name} function is missing.");
     auto api_error_result = ${fname}(${args});
     if (api_error_result != ${success})
-        throw std::runtime_error("Error in ${name}.");
+        throw std::runtime_error("Error in ${name} of: " + xobject.name);
```

Comment:
There is no guarantee that `xobject` will have a `name` member field. It works for custom op, but future interfaces may not have a `name` member field. You should use `typeid` instead: `typeid(object_ptr.data)` or add a method to `manage_generic_ptr`

Reply:
But wouldn't the `typeid` return some mangled name ?  for now i find using `xobject.name` clearer.  If we change interface in future, we can change this one too.

Reply:
> But wouldn't the typeid return some mangled name ?

I think it is for gcc and clang unfortunately. We could try to store the type name in the `manage_generic_ptr` class(as computed from [here](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/develop/src/include/migraphx/type_name.hpp#L34) which will need to copy into migaphx.hpp), but we need to update api.py to add the typename parameters along with current copy constructor and destructor parameters:

https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/develop/tools/api.py#L1106

And then pass the typename to `make_handle` here:

https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/develop/src/api/include/migraphx/migraphx.hpp#L344

> for now i find using xobject.name clearer.

Interfaces dont provide a `.name` so we can't use it even if it is clearer.

> If we change interface in future, we can change this one too.

`.name` donesn't come from the interface, its an implementation detail of custom op, which is the problem. This will prevent us from adding interfaces in the future as they will fail to compile since they would be missing `.name` in its implementation.

Reply:
It prints `type` of the custom_op struct not really name of the custom_op though.